tests/components/datadog/test_init.py

```diff
@@ -160,17 +149,17 @@ def test_state_changed(self, mock_connection, mock_client):
                     ]
                 )
 
-            assert mock_client.gauge.call_args == mock.call(
+            assert mock_statsd.gauge.call_args == mock.call(
                 "ha.sensor",
                 out,
                 sample_rate=1,
                 tags=[f"entity:{state.entity_id}"],
             )
 
-            mock_client.gauge.reset_mock()
+            mock_statsd.gauge.reset_mock()
 
         for invalid in ("foo", "", object):
             handler_method(
                 mock.MagicMock(data={"new_state": ha.State("domain.test", invalid, {})})
             )
-            assert not mock_client.gauge.called
+            assert not mock_statsd.gauge.called
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)